drivers/bridge/bridge.go

```diff
@@ -1451,3 +1479,60 @@ func electMacAddress(epConfig *endpointConfiguration, ip net.IP) net.HardwareAdd
 	}
 	return netutils.GenerateMACFromIP(ip)
 }
+
+func (d *driver) RestoreBridgeEndpoint(n *bridgeNetwork, endpoint *bridgeEndpoint, iface driverapi.InterfaceInfo) error {
+	ipv4Addr := iface.Address()
+	// Restore ipv4
+	_, err := ipAllocator.RequestIP(n.bridge.bridgeIPv4, ipv4Addr.IP)
+	if err != nil {
+		return err
+	}
+
+	n.Lock()
+	config := n.config
+	n.Unlock()
+
+	// Restore ipv6
+	ipv6Addr := iface.AddressIPv6()
+	if ipv6Addr.IP != nil && len(ipv6Addr.IP) != 0 {
+		network := n.bridge.bridgeIPv6
+		if config.FixedCIDRv6 != nil {
+			network = config.FixedCIDRv6
+		}
+		_, err := ipAllocator.RequestIP(network, ipv6Addr.IP)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Restore container interface name
+	sbox, err := linkByMacAddress(iface.MacAddress().String())
+	if err != nil {
+		return err
```

Comment:
@chenchun that is okay. can you please make sure the basic use-case works fine in 1.9 without any of your hot-upgrade changes ? That way you don't have to worry about master branch in docker out-of-date for your needs. I do understand that if you want to also include the hot-restart testing in the same branch, then it might be difficult for you.
Please note that we are prioritizing the PR  for the reason that it is having a positive impact on the release. If it impacts the base docker release code, we wont be able to accept it till it is resolved properly. Hope you understand that.